x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -1407,6 +1407,22 @@ public void testParamMixed() {
         );
     }
 
+    public void testIntervalParam() {
+        LogicalPlan stm = statement(
+            "row x = ?1::datetime | eval y = ?1::datetime + ?2::date_period",
```

Comment:
nit: I don't think this test is testing a new kind of statement to be parsed, and is kinda duplicated in the `10_basic.yml` test.

The interesting bit here is that `?1` is used twice. Maybe we should just make the test case in `10_basic.yml` also re-use one of the parameters, and then remove this test here.

Reply:
I'd like to keep this test here, it is not exactly a duplicate of the integration test in `10_basic.yml`. The major difference is that the test in `10_basic.yml` uses named parameter, and `testIntervalParam` uses positional parameter. `?1` is used twice intentionally to make sure we support this kind of usage.